app/build.gradle

```diff
@@ -45,10 +45,8 @@ dependencies {
     testImplementation 'junit:junit:4.12'
     androidTestImplementation 'androidx.test.ext:junit:1.1.0'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
-    androidTestImplementation 'com.android.support.test:rules:1.1.0'
-    androidTestImplementation 'com.android.support.test:runner:1.1.0'
-    androidTestImplementation 'com.android.support.test.uiautomator:uiautomator-v18:2.1.1'
-    androidTestImplementation 'org.hamcrest:hamcrest-library:1.3'
```

Comment:
Should these be kept?

Reply:
this looks like dependencies for running tests so yes.